src/Common/src/System/IO/Win32Marshal.cs

```diff
@@ -99,11 +99,13 @@ internal static Exception GetExceptionForWin32Error(int errorCode, string path)
         }
 
         /// <summary>
-        ///     Returns a HRESULT for the specified Win32 error code.
+        /// If not already an HRESULT, returns an HRESULT for the specified Win32 error code.
         /// </summary>
         internal static int MakeHRFromErrorCode(int errorCode)
         {
-            Debug.Assert((0xFFFF0000 & errorCode) == 0, "This is an HRESULT, not an error code!");
+            // Don't convert it if it is already an HRESULT
+            if ((0xFFFF0000 & errorCode) == 0)
```

Comment:
I'm missing something here, wont all HRESULTS result in != 0 because they have high bits set? eg 0xFFFF0000 & 0x80070005 is not zero and it's an HRESULT.